src/include/miopen/graphapi/graphapi.hpp

```diff
@@ -49,6 +51,7 @@ class BackendDescriptor : public miopenBackendDescriptor
                               int64_t* elementCount,
                               void* arrayOfElements) = 0;
     virtual void execute(miopenHandle_t handle, miopenBackendDescriptor_t variantPack);
+    virtual Operation* getOperation();
```

Comment:
Not all `BackendDescriptors` are `Operation Nodes`, so I'm not sure if this method belongs here.

Reply:
I partially agree with you.

My point is the following. Graph API operates on pointers and doesn't know the type of the descriptor. The base class implementation returns `nullptr`. A descendant class who is an operation descriptor should override it and return a valid pointer.

Another option is to define a separate base class for all Operations and move `getOperation()` there. It will cost an additional dynamic_cast:

```c++
class BackendOperationGraphDescriptor : public BackendDescriptor
{
public:
    void setAttribute(...) override
    {
        BackendDescriptor* pBaseDescr    = deref(static_cast<miopenBackendDescriptor_t*>(arrayOfElements));
        auto& baseDescr = deref(pBaseDescr);
        auto& opDescr = dynamic_cast<ProposedBaseForOperations&>(baseDescr);
        opGraphBuilder.setOperation(opDescr.getOperation());
    }
};
``` 

Seems like not a big price for a good logic. I'll move the function.

Reply:
may be we should leave it as is for now and come back to it in the near future.